tests/queries/0_stateless/02949_parallel_replicas_in_subquery.sql

```diff
@@ -8,19 +8,23 @@ INSERT INTO merge_tree_in_subqueries VALUES(5, 'test5', 0);
 
 SET max_parallel_replicas=3, cluster_for_parallel_replicas='test_cluster_one_shard_three_replicas_localhost', parallel_replicas_for_non_replicated_merge_tree=1;
 
-SELECT * FROM merge_tree_in_subqueries WHERE id IN (SELECT * FROM system.numbers LIMIT 0) SETTINGS allow_experimental_parallel_reading_from_replicas=2; -- { serverError SUPPORT_IS_DISABLED }
+SELECT * FROM merge_tree_in_subqueries WHERE id IN (SELECT * FROM system.numbers LIMIT 0) SETTINGS allow_experimental_parallel_reading_from_replicas=2, parallel_replicas_allow_subqueries_for_in=0; -- { serverError SUPPORT_IS_DISABLED }
```

Comment:
I'd suggest adding a test where both tables are mergetree's and both have different partition keys. It's how most of the bugs have appeared.

Reply:
Interesting how MergeTree partitioning key can be related to some issues here

Reply:
It does not trigger bugs, but it helps detecting them:

* If both have the same partition key. Most likely, in testing, both tables have no partition key and likely one or zero parts. The part is called `all_1_1_0` in both tables. If there is a bug in the coordinator when working with multiple tables we won't detect it because assigning `tableA.all_1_1_0` to the coordinator of tableB is ok.

* If both tables have a different partition key, then a bug in the coordination will be immediately obvious since the assigned part won't exists / can't match the other table.